studiopress/genesis-sample
# Switch to custom script to zip files for production

This PR switches from `git archive` to a custom npm script to zip theme files for production. This overcomes two issues:

1. The Path Finder app for macOS unzips files created with `git archive -o genesis-sample.zip --prefix=genesis-sample/ HEAD` into `genesis-sample.zip Folder/genesis-sample` instead of `genesis-sample`. (This bug has been reported and acknowledged, but we don't know when it will be fixed.) The issue was reported to us by @hellofromtonya.

2. Avoids having to edit the zip file in-place after creation to rename .md files to .txt. The `md-to-txt.js` script (removed in this PR) intermittently corrupts zip files due to what seems to be a [longstanding bug in the adm-zip library](https://github.com/cthackers/adm-zip/issues/25). This PR renames the .md files to .txt ones prior to zip instead of editing the zip file in-place, and uses the archiver library for compression instead of adm-zip.

Other benefits:
- Gives us more flexibility if we need to adjust the export process in future.
- Removes the `export-ignore` lines in `.gitattributes`, which cause named files to be omitted if people download the zip directly from GitHub using the “Download zip” link under the clone button the code tab.

To test:

- Run `npm install`
- Run `npm run zip`

Confirm that the resulting zip file includes only relevant (non-development) files, and that it unzips into a `genesis-sample` folder.
